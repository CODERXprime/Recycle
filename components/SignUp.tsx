import * as React from 'react';
import { Component } from 'react';
import { connect } from 'react-redux';
import { AppRegistry, View, TextInput, Text } from 'react-native';
import {
  FormLabel,
  FormInput,
  FormValidationMessage,
  Button,
  Avatar,
  Card,
  Divider,
} from 'react-native-elements';
import firebase from '../firebase';

const db = firebase.firestore();

const myUsers = db.collection('User');
interface State {
  userName: string;
  email: string;
  password: string;
  points: number;
  loggedIn: boolean;
  uid: string;
}
interface Props {}

export default class SignUp extends Component<Props, State> {
  constructor(props: Props, context?: any) {
    super(props, context);
    this.state = {
      userName: '',
      email: '',
      password: '',
      points: 0,
      loggedIn: false,
      uid: '',
    };
  }
  componentDidMount() {
    this.inputOne.focus();
    this.inputTwo.focus();
    this.inputThree.focus();
  }
  signUp = (email, password, userName) => {
    try {
      firebase
        .auth()
        .createUserWithEmailAndPassword(email, password)
        .then(user =>
          db
            .collection('User')
            .doc(user.user.uid)
            .set({
              userName: userName,
              email: email,
              password: password,
              id: `${user.user.uid}`,
              points: 0,
            })
        );
      this.clear();
    } catch (error) {
      console.log(error.toString());
    }
  };

  clear = () => {
    this.inputOne.clearText();
    this.inputTwo.clearText();
    this.inputThree.clearText();
    this.setState({
      loggedIn: true,
    });
  };

  logOut = () => {
    try {
      firebase.auth().signOut();
      this.setState({
        userName: '',
        email: '',
        password: '',
        points: 0,
        loggedIn: false,
      });
    } catch (error) {
      console.log(error.toString());
    }
  };

  logIn = (email, password) => {
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(user => db.collection('User').doc(user.user.uid))
      .then(query => query.get())
      .then(found => found.data())
      .then(data => {
        this.setState({
          userName: data.userName,
          email: data.email,
          points: data.points,
          loggedIn: true,
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
  };
  private inputOne: FormInput;
  private inputTwo: FormInput;
  private inputThree: FormInput;
  render() {
    return (
      <View>
     config-app.json-for-deployment
        {this.state.loggedIn === false ? (
          <View>
            <FormLabel>User Name</FormLabel>
            <FormInput
              onChangeText={userName => this.setState({ userName })}
              ref={inputOne => (this.inputOne = inputOne)}
            />
            <FormLabel>Email</FormLabel>
            <FormInput
              onChangeText={email => this.setState({ email })}
              ref={inputTwo => (this.inputTwo = inputTwo)}
            />
            <FormLabel>Password</FormLabel>
            <FormInput
              secureTextEntry
              onChangeText={password => this.setState({ password })}
              ref={inputThree => (this.inputThree = inputThree)}
            />
            <FormLabel />
            <Button
              raised
              backgroundColor="#3E9428"
              title="Sign Up"
              onPress={() =>
                this.signUp(
                  this.state.email,
                  this.state.password,
                  this.state.userName
                )
              }
            />
            <Button
              style={{ marginTop: 10 }}
              raised
              backgroundColor="#3E9428"
              title="Log In"
              onPress={() => this.logIn(this.state.email, this.state.password)}
            />
          </View>
        ) : (
          <View>
            <Card containerStyle={{ marginTop: 150 }}>
              <Avatar
                xlarge
                title={this.state.userName[0]}
                containerStyle={{ marginLeft: 100, marginRight: 115 }}
                rounded
              />
              <Text
                style={{
                  marginTop: 25,
                  marginLeft: 110,
                  marginRight: 75,
                  fontSize: 40,
                }}
              >
                {this.state.userName}
              </Text>
              <Text
                style={{
                  marginTop: 10,
                  marginLeft: 110,
                  marginRight: 75,
                  fontSize: 35,
                }}
              >
                Points: {this.state.points}
              </Text>
              <Button
                style={{ marginTop: 10 }}
                raised
                backgroundColor="#3E9428"
                title="Log Out"
                onPress={() => this.logOut()}
              />
            </Card>
          </View>
        )}
      </View>
    );
  }
}

//export default connect(null)(SignUp);
