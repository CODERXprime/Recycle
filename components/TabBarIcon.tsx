import * as React from 'react';
import { Component } from 'react';
import { Ionicons } from '@expo/vector-icons';
import Colors from '../constants/Colors';

interface TabBarIconProps {
  name: string;
  focused: string;
}

export default class TabBarIcon extends Component<TabBarIconProps> {
  render() {
    return (
      <Ionicons
        name={this.props.name}
        size={26}
        style={{ marginBottom: -3 }}
        color={
          this.props.focused ? Colors.tabIconSelected : Colors.tabIconDefault
        }
      />
    );
  }
}
