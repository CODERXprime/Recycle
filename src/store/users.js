import axios from 'axios';

const userReducer = (state = [], action) => {
  return state;
};

export default userReducer;
