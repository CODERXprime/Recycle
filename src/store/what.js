/*
FETCHING DATA FROM CLOUD VISION
GET http://
*/

import axios from 'axios';

const whatReducer = (state = [], action) => {
  return state;
};

export default whatReducer;
