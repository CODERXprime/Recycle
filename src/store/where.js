/*
FETCHING DATA FROM EARTH911 API in JSON
GET http://api.earth911.com/earth911.methodName
http://api.earth911.com/
*/

import axios from 'axios';

const FIND_PLACES_TO_RECYCLE = 'FIND_PLACES_TO_RECYCLE';

const _findPlacesToRecycle = locationDetails => ({ type: FIND_PLACES_TO_RECYCLE, locationDetails });

const initialState = [];

const whereReducer = (state = initialState, action) => {
  switch (action.type) {
    case FIND_PLACES_TO_RECYCLE:
      return action.locationDetails;
    default: return state;
  };
};

export const findPlacesToRecycle = (api_key, GPS, productInfo) => dispatch => {
  console.log(productInfo)
  /*
  1) GET MATERIAL ID: materials = api.earth911.searchMaterials({
    'api_key': api_key,
    'query': 'used motor oil',
})
  2) {We may be able to skip this call with GPS coordinates from cell phone}
  Convert Zip Code to Lat/Long: postal_data = api.earth911.getPostalData({
    'api_key': api_key,
    'country': 'US',
    'postal_code': '90210',
})
  3) With Lat/Long, can now find what locations accept certain recycables
    results = api.earth911.searchLocations({
   'api_key': api_key,
   'latitude': 34.09,
   'longitude': -118.41,
   'material_id': [1, 4],
   'max_distance': 5,
})

len(results) >>> 57

RES:
{'curbside': False,
 'description': 'Jim Falk Lexus of Beverly Hills',
 'distance': 1.7,
 'latitude': 34.07,
 'location_id': '2VwAkLGZ',
 'location_type_id': 0,
 'longitude': -118.39,
 'municipal': False}

  4) Expand on location details
  details = api.earth911.getLocationDetails({
  'api_key': api_key,
  'location_id': '2VwAkLGZ',
})

RES:
{'2VwAkLGZ': {'address': '9230 Wilshire Boulevard',
              'city': 'BEVERLY HILLS',
              'country': 'US',
              'created': '2007-12-18T16:03:00',
              'curbside': False,
              'description': 'Jim Falk Lexus of Beverly Hills',
              'event_only': False,
              'fax': '',
              'flag': 'none',
              'geocoded': True,
              'hours': 'Please call for hours of operation.',
              'latitude': 34.067096470199999,
              'location_type_id': 0,
              'longitude': -118.39313399700001,
              'materials': [{'business': False,
                             'business_method': 'none',
                             'description': 'Used Motor Oil',
                             'dropoff': True,
                             'material_id': 1,
                             'notes': '',
                             'pickup': False,
                             'residential': True,
                             'residential_method': 'dropoff'}],
              'municipal': False,
              'national': False,
              'notes': '',
              'phone': '(310) 274-5200',
              'postal_code': '90212',
              'province': 'CA',
              'region': 'LOS ANGELES',
              'updated': '2007-12-21T11:23:00',
              'url': ''}}
  */

  materials = api.earth911.searchMaterials({
    //this will have to talk to './what.js' in that the query is populated by the vision component.
    api_key,
    'query': 'used motor oil',
  });


  results = api.earth911.searchLocations({
    api_key,
    'latitude': GPS.latitude,
    'longitude': GPS.longitude,
    //get lat and long from phone GPS
    'material_id': [1, 4],
    //params below are optional
    'max_distance': 5,
    'max_results': 10,
  });

  locationDetails = api.earth911.getLocationDetails({
    api_key,
    'location_id': '2VwAkLGZ',
    //send this back to RecPlacesCard.js
  });

  //probably break this out into multiple thunks

  return axios
    //template until we get API KEY
    .get(`http://api.earth911.com/earth911.methodName?key=${api_key}&productInfo=${productInfo}`)
    .then(res => res.data)
    .then(locationDetails => dispatch(_findPlacesToRecycle(locationDetails)));
};

export default whereReducer;
