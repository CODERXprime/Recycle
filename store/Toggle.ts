const TOGGLE_ON: string = 'TOGGLE_ON';
const _toggleOn = () => ({ type: TOGGLE_ON })
export const toggleOn = () => dispatch => dispatch(_toggleOn());

const toggleReducer = (state = false, action) => {
  switch (action.type) {
    case TOGGLE_ON:
      return true
    default:
      return state;
  };
};

export default toggleReducer;
