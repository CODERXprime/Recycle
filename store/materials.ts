import axios from 'axios';
const baseURL: string = 'http://api.earth911.com/earth911.';

//ACTIONS
const FIND_MATERIALS: string = 'FIND_MATERIALS';
const GET_MATERIAL_DETAIL: string = 'GET_MATERIAL_DETAIL';

//ACTION CREATORS
const _findMaterials = (materials) => ({ type: FIND_MATERIALS, materials });
const _getMaterialDetail = (material) => ({ type: GET_MATERIAL_DETAIL, material });

//REDUCER
const initialState = {
  foundMaterials: [],
  materialDetails: {
    // description: '',
    // image: '',
    // long_description: '',
    // material_id: '',
  },
}
const materialsReducer = (state = initialState, action) => {
  switch (action.type) {
    case FIND_MATERIALS:
      return { ...state, foundMaterials: action.materials };
    case GET_MATERIAL_DETAIL:
      return { ...state, materialDetails: action.material }
    default: return state;
  };
};

//THUNKS
export const getMaterialDetail = (api_key: string, materialId: string) => dispatch => {
  return axios
    .get(`${baseURL}getMaterials?api_key=${api_key}`)
    .then(res => res.data)
    .then(materials => {
      return materials.result.find(material => {
        return material.material_id === materialId;
      })
    })
    .then(material => dispatch(_getMaterialDetail(material)))

  /* SAMPLE RES {
"description": "#1 Plastic Bags",
"url": "",
"description_legacy": "",
"material_id": 445,
"long_description": "Plastic bags are used to transport products or to seal foods. #1 Plastic bags may be difficult to recycle because they have limited markets.",
"family_ids": [
9,
106,
108
],
"image": "materials/1-plastic-bags.jpg"
}, */
};

export const searchMaterials = (api_key: string, materialSearch: string) => dispatch => {
  return axios
    .get(`${baseURL}searchMaterials?api_key=${api_key}&query=${materialSearch}`)
    .then(res => res.data)
    .then(materials => {
      if (!materials.result.length) {
        materials.result.description = 'Not Recycable, but maybe you can try Reusing it!';
        materials.result.material_id = 10000;
      }
      dispatch(_findMaterials(materials.result))
    })
    .catch(error => {
      console.log('materials.ts');
      if (error.response) {
          /*
           * The request was made and the server responded with a
           * status code that falls out of the range of 2xx
           */
          console.log(error.response.data);
          console.log(error.response.status);
          console.log(error.response.headers);
      } else if (error.request) {
          /*
           * The request was made but no response was received, `error.request`
           * is an instance of XMLHttpRequest in the browser and an instance
           * of http.ClientRequest in Node.js
           */
          console.log(error.request);
      } else {
          // Something happened in setting up the request and triggered an Error
          console.log('Error', error.message);
      }
      console.log(error.config);
    });
  /* SAMPLE RES {
  "num_results": 1,
  "result": [
  {
  "url": "",
  "exact": true,
  "description": "Toothbrushes",
  "material_id": 587
  }
  ]}
  */
}

export default materialsReducer;
