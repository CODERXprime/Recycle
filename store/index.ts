import { createStore, applyMiddleware, combineReducers } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

import what from './what';
import where from './where';
import users from './users';
import materials from './materials';
import toggle from './toggle';

export interface reducerLayout {
  state: string[],
  action: object | null,
};

const reducer = combineReducers({
  what,
  where,
  users,
  materials,
  toggle,
});

export default createStore(reducer, applyMiddleware(logger, thunk));
