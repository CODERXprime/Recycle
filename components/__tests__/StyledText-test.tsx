import 'react-native';
import * as React from 'react';
import { MonoText } from '../StyledText';
import TabBarIcon from '../TabBarIcon';
import SignUp from '../SignUp';
import { Provider } from 'react-redux';
import store from '../../store/index';
import * as renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<MonoText>Snapshot test!</MonoText>).toJSON();
  expect(tree).toMatchSnapshot();
});
it('TabBarIcon renders correctly', () => {
  const tree = renderer.create(<TabBarIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});
it('SignUp renders correctly', () => {
  const tree = renderer.create(
    <Provider store={ store }>
      <SignUp />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})