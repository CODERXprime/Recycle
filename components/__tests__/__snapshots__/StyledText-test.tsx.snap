// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        User Name
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        Email
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        Password
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#3E9428",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "marginLeft": 15,
            "marginRight": 15,
          },
          Object {
            "shadowColor": "rgba(0,0,0, .4)",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 1,
            "shadowRadius": 1,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginTop": 10,
          }
        }
      >
        <View
          pointerEvents="box-only"
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#9E9E9E",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 19,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "backgroundColor": "#3E9428",
              },
              undefined,
              Object {
                "padding": 12,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {},
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontSize": 17.5,
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Log In
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`TabBarIcon renders correctly 1`] = `<Text />`;

exports[`renders correctly 1`] = `
<Text
  style={
    Array [
      undefined,
      Object {
        "fontFamily": "space-mono",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;
